addons/web_editor/static/src/js/common/utils.js

```diff
@@ -487,6 +488,39 @@ function _isMobileView(targetEl) {
 function _getLinkLabel(linkEl) {
     return linkEl.innerText.trim().replaceAll("\u200B", "");
 }
+/**
+ * Finds the color in the given array that is closest to the target color.
+ * @param {string} targetColor - The target color in hex string format (#RRGGBB).
+ * @param {string[]} colorArray - An array of colors to compare against the target color.
```

Comment:
From my discussion with Luc, it came out that
* Option 1 is the best option if not technically demanding (This way it would work in all cases)
* Otherwise, the initial idea was to change the color(s) **in the template(s)** to one(s) covered by the "solid" colors.

Of course don't hesitate to ask if you have any question :)

Reply:
@flch-odoo Done changes :+1:

Reply:
@visp-odoo Thanks for the updates! I just tested and it seems that the switch only works the first time. After opening the picker on an item, if you go see another block then come back to the first item and open "Background Color", we end up on the "solid" tab this time.

Reply:
@flch-odoo I just missed that earlier :sweat_smile:  .I have made some changes.I test it.IMO, now it works fine. Please take a look into this.